nmagent/internal/wireserver.go

```diff
@@ -114,17 +116,25 @@ func (w *WireserverTransport) RoundTrip(inReq *http.Request) (*http.Response, er
 			// Wireserver gets angry when there's no Content-Length header, and returns
 			// a 411 status code:
 			req.ContentLength = int64(len(emptyJSONString))
-
-			// the Content-Type must also be manually set, because the net/http
-			// Content-Type detection can't be run without draining the Reader set
-			// for the Request Body. This is a good idea anyway, because the
-			// detection uses heuristic methods to figure out the Content-Type, which
-			// may not necessarily be correct for a short JSON string like this:
-			req.Header.Set(HeaderContentType, MimeJSON)
 		}
+
+		// the Content-Type must also be manually set, because the net/http
+		// Content-Type detection can't be run without draining the Reader set
+		// for the Request Body. This is a good idea anyway, because the
+		// detection uses heuristic methods to figure out the Content-Type, which
+		// may not necessarily be correct for a short JSON string like this:
+		req.Header.Set(HeaderContentType, MimeJSON)
 	}
 
 	// execute the request to the downstream transport
+	req2 := req.Clone(context.TODO())
+	logger.Printf("internal req headers: %v", req2.Header)
+	logger.Printf("req2 body unread: %v", req2.Body)
+	req2BodyBytes, _ := io.ReadAll(req2.Body)
+
+	logger.Printf("internal req body bytes: %v", string(req2BodyBytes))
```

Comment:
some wireserver calls will be GETs, which won't have a body. let's make this optional only if `req2.Body` is non-nil